falcon-solver/src/main/java/pt/haslab/causalSolver/CausalSolver.java

```diff
@@ -535,17 +541,20 @@ public static void outputCausalOrderJSON()
         TreeSet<Event> orderedEvents = new TreeSet<Event>( allEvents.values() );
         try
         {
-            FileWriter outfile = new FileWriter( new File( props.getProperty( Parameters.OUTPUT.toString() ) ) );
+            String outputFile = props.getProperty( Parameters.OUTPUT.toString() );
+            FileWriter fileWriter = new FileWriter( new File( props.getProperty( Parameters.OUTPUT.toString() ) ) );
             JSONArray jsonEvents = new JSONArray();
             for ( Event e : orderedEvents )
             {
                 JSONObject json = e.toJSONObject();
-                System.out.println( json.toString() );
+                logger.debug( json.toString() );
```

Comment:
Add the conditional statement: `if (logger.isDebugEnabled())` as this is a expensive operation.